optimade/client/client.py

```diff
@@ -93,7 +91,7 @@ class OptimadeClient:
 
     def __init__(
         self,
-        base_urls: Union[None, AnyUrl, List[AnyUrl]] = None,
+        base_urls: Optional[Union[str, List[str]]] = None,
```

Comment:
On line 58 you defined base_urls as an Iterable yet here you define it as a list.
This is why mypy gives an error on lines 115 and 121
```suggestion
        base_urls: Optional[Union[str, Iterable[str]]] = None,
```